docs/walkthrough-1.7.md

```diff
@@ -0,0 +1,489 @@
+- `ServiceClass`: A *type* of managed service offered by a particular broker.
+Each time a new `Broker` resource is added to the cluster, the service catalog
+controller connects to the corresponding broker server to obtain a list of
+service offerings. A new `ServiceClass` resource will automatically be created
+for each.
+- `Instance`: A provisioned instance of a `ServiceClass`. These are created
+by cluster users who wish to make a new concrete _instance_ of some _type_ of
+managed service to make that available for use by one or more in-cluster
+applications. When a new `Instance` resource is created, the service catalog
+controller will connect to the appropriate broker server and instruct it to
+provision the service instance.
+- `Binding`: A "binding" to an `Instance`. These are created by cluster users
+who wish for their applications to make use of a service `Instance`. Upon
+creation, the service catalog controller will create a Kubernetes `Secret`
+containing connection details and credentials for the service instance. Such
+`Secret`s can be mounted into pods as usual.
+
+These concepts and resources are the building blocks of the service catalog.
+
+## Step 3 - Installing the UPS Broker
+
+In order to effectively demonstrate the service catalog, we will require a
+sample broker server. To proceed, we will deploy the [User Provided Service
+broker (hereafter, "UPS")](https://github.com/kubernetes-incubator/service-catalog/tree/master/contrib/pkg/broker/user_provided/controller)
+to our own Kubernetes cluster. Similar to the service catalog system itself,
+this is easily installed using a provided Helm chart. The chart supports a
+wide variety of customizations which are detailed in that directory's
+[README.md](https://github.com/kubernetes-incubator/service-catalog/blob/master/charts/ups-broker/README.md).
+
+**Note:** The UPS broker emulates user-provided services as they exist in
+Cloud Foundry. Essentially, values provided during provisioning are merely
+echoed during binding. (i.e. The values *are* the service.) This is a trivial
+broker server but it's deliberately employed in this walkthrough to avoid
+getting hung up on the distracting details of some other technology.
+
+To install with defaults:
+
+```console
+helm install charts/ups-broker --name ups-broker --namespace ups-broker
+```
+
+## Step 4 - Installing the Latest `kubectl`
+
+As with Kubernetes itself, interaction with the service catalog system is
+achieved through the `kubectl` command line interface. Chances are high that
+you already have this installed, however, the service catalog *requires*
+`kubectl` version 1.7 or newer.
+
+To proceed, we must download and install `kubectl` version 1.7 or newer. Do so by following the 
+[standard instructions](https://kubernetes.io/docs/tasks/kubectl/install/).
+
+For example, on a mac,
+```console
+curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.7.0/bin/darwin/amd64/kubectl
```

Comment:
`dl.k8s.io` points to `storage.googleapis.com/kubernetes-release`, can use that instead...